src/clj/rems/api/services/blacklist.clj

```diff
@@ -0,0 +1,51 @@
+(ns rems.api.services.blacklist
+  (:require [clj-time.core :as time]
+            [rems.application.events :as events]
+            [rems.db.blacklist :as blacklist]
+            [rems.db.core :as db]
+            [rems.db.resource :as resource]
+            [rems.db.users :as users]
+            [rems.json :as json]
+            [schema.core :as s]))
+
+(defn- event->json [event]
+  (s/validate blacklist/BlacklistEvent event)
+  (json/generate-string event))
+
+(defn- check-foreign-keys [event]
+  ;; TODO: These checks could be moved to the database as (1) constraint checks or (2) fields with foreign keys.
+  (when-not (users/user-exists? (:userid event))
+    (throw (IllegalArgumentException. "user doesn't exist")))
+  (when-not (resource/ext-id-exists? (:resource/ext-id event))
+    (throw (IllegalArgumentException. "resource doesn't exist")))
+  event)
+
+(defn- command->event [command user-id]
```

Comment:
the `user-id` parameter should probably be called `actor` here and elsewhere, otherwise it sounds like it is the user that is added to the blacklist

Reply:
Done.